File: projects/agentic-automation-framework/src/agent/engine.js
import fs from "fs";
import path from "path";
import { runWorkflow } from "../playwright/runner.js";
import { sendTelegramMessage } from "../channels/telegram.js";
import { sendWhatsAppMessage } from "../channels/whatsapp.js";
import { env } from "../config/env.js";

export function loadWorkflow(filePath) {
  const p = path.resolve(filePath);
  return JSON.parse(fs.readFileSync(p, "utf-8"));
}

export async function executeWorkflowFile(filePath, { notify = false } = {}) {
  const workflow = loadWorkflow(filePath);
  const result = await runWorkflow(workflow, { headless: env.headless });

  if (notify) {
    const text = result.ok
      ? `✅ [${result.name}] Passed in ${result.durationMs}ms\nURL: ${result.finalUrl}`
      : `❌ [${result.name}] Failed in ${result.durationMs}ms\nError: ${result.error}`;

    await Promise.allSettled([
      sendTelegramMessage({ token: env.telegram.token, chatId: env.telegram.chatId, text }),
      sendWhatsAppMessage({ token: env.whatsapp.token, phoneNumberId: env.whatsapp.phoneNumberId, to: env.whatsapp.to, text })
    ]);
  }

  return result;
}

File: projects/agentic-automation-framework-python/agentic/agent/engine.py
import json
from pathlib import Path
from agentic.config import CONFIG
from agentic.playwright_runner import run_workflow
from agentic.channels.telegram import send_message as send_tg
from agentic.channels.whatsapp import send_message as send_wa


def load_workflow(file_path: str):
    return json.loads(Path(file_path).read_text())


def execute_workflow_file(file_path: str, notify=False):
    wf = load_workflow(file_path)
    result = run_workflow(wf, headless=CONFIG["headless"])

    if notify:
        text = (
            f"✅ [{result['name']}] Passed in {result['durationMs']}ms\nURL: {result.get('finalUrl','')}"
            if result.get("ok")
            else f"❌ [{result['name']}] Failed in {result['durationMs']}ms\nError: {result.get('error','')}"
        )
        try:
            send_tg(CONFIG["telegram"]["token"], CONFIG["telegram"]["chat_id"], text)
        except Exception:
            pass
        try:
            send_wa(CONFIG["whatsapp"]["token"], CONFIG["whatsapp"]["phone_number_id"], CONFIG["whatsapp"]["to"], text)
        except Exception:
            pass

    return result
